commit 1a23127c6d0676877225233d83f5de551c4d80a6
Author: Steve French <smfrench@gmail.com>
Date:   Tue Sep 19 18:40:03 2017 -0500

    SMB3: Warn user if trying to sign connection that authenticated as guest
    
    commit c721c38957fb19982416f6be71aae7b30630d83b upstream.
    
    It can be confusing if user ends up authenticated as guest but they
    requested signing (server will return error validating signed packets)
    so add log message for this.
    
    Signed-off-by: Steve French <smfrench@gmail.com>
    Reviewed-by: Ronnie Sahlberg <lsahlber@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/fs/cifs/smb2pdu.c b/fs/cifs/smb2pdu.c
index 9d8ad4b285fa..b98d96a41518 100644
--- a/fs/cifs/smb2pdu.c
+++ b/fs/cifs/smb2pdu.c
@@ -1010,6 +1010,8 @@ SMB2_sess_setup(const unsigned int xid, struct cifs_ses *ses,
 	while (sess_data->func)
 		sess_data->func(sess_data);
 
+	if ((ses->session_flags & SMB2_SESSION_FLAG_IS_GUEST) && (ses->sign))
+		cifs_dbg(VFS, "signing requested but authenticated as guest\n");
 	rc = sess_data->result;
 out:
 	kfree(sess_data);
